e2e/test/PlugAndPlayTests.cs

```diff
@@ -0,0 +1,332 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    /// <summary>
+    /// Test class containing all tests to be run for plug and play.
+    /// </summary>
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    [TestCategory("PlugAndPlay")]
+    public class PlugAndPlayTests
+    {
+        private const string DevicePrefix = "plugAndPlayDevice";
+        private const string ModulePrefix = "plugAndPlayModule";
+        private const string TestModelId = "dtmi:com:example:testModel;1";
+
+        private static TestLogging s_log = TestLogging.GetInstance();
+        private readonly ConsoleEventListener _listener = TestConfig.StartEventListener();
+
+        [TestMethod]
```

Comment:
Decided against using `DataTestMethod`, since data driven tests cannot be parallelized using MS test: https://github.com/Microsoft/testfx-docs/blob/master/RFCs/004-In-Assembly-Parallel-Execution.md